### PR TITLE
Url validation

### LINKED ISSUE #4: Image URL Validation
#### As a user I want to be prevented from submitting invalid image URLs.

Acceptance criteria:
- [x] I cannot successfully save an image with an invalid URL.
- [x] An error message is associated with the appropriate input field on
  failure.

Discussion Topic:
- [ ] What is a valid image URL?

Dependencies:
- Save Image Link

### PR DESCRIPTION
**Pull request by jeremyjpark**
Closes #4

### CODE REVIEW

File: test/models/image_test.rb [hunk 1]
```diff
@@ -1,11 +1,26 @@
...
+    assert Image.new(url: 'http://www.google.com').valid?
+  end
 
-    image = Image.new(url: 'www.google.com')
-    assert_predicate image, :valid?
+  def test_image_url__bad_url_rescued
+    assert_nothing_raised do
+      Image.new(url: 'invalid url')
+      Image.new(url: 'ftp://www.google.com')
+    end
```
Comment (scuendet) [inline test/models/image_test.rb]: What's this test for?

**Comment by scuendet** [comment]
Left a minor comment, looks good overall though!

File: test/models/image_test.rb (hunk 1 above)
Comment (jeremyjpark) [inline test/models/image_test.rb]: I realized that there needs to be a test for making sure that a URI::InvalidURIError gets caught.

Comment (scuendet) [inline test/models/image_test.rb]: But it's internal to your method, so I don't think you actually need it.

**Pull request merged**